### ISSUE TITLE
389-ds-base fails constantly in docker FreeIPA environment

### ISSUE THREAD

**Issue by abbra** [389ds/389-ds-base#5942]
**Issue Description**
FreeIPA gating in Azure CI uses Docker containers. In one specific test that runs upgrade of IPA environment 389-ds regularly fails due to detecting too little memory and then refuses to operate. The tests starts and restarts IPA setup, including restarts of the directory server instance.

It doesn't happen every time and in other tests, sometimes I am able to pass the test after 4-5 retries but today I stuck with 14 retries with no success.

**Package Version and Platform:**
 - Platform: Fedora 38
 - Package and version: 389-ds-base-2.3.7-1.fc38.x86_64

I attached logs from one of failing tests.
[test_integration-test_upgrade.py-TestUpgrade-test_krb_uri_txt_to_cname.tar.gz](https://github.com/389ds/389-ds-base/files/12766163/test_integration-test_upgrade.py-TestUpgrade-test_krb_uri_txt_to_cname.tar.gz)

**Comment by progier389** [comment]
Hi, IMHO the issue could be anything. From what you describe the memory is very low so any change to any component could have been enough to eat some memory and trigger the issue. 
Your best change is to try to tweak the configuration to have enough memory
I will start by reducing the connection table size by reducing the maximum number of file descriptos 
 (NOFILE hard limit from 100K to 1024 ) 
Should apply  https://access.redhat.com/solutions/1257953 to dirsrv@instancename services ... (or more globally to the whole system ...)

**Comment by abbra** [comment]
@progier389 please look into the attached tarball. The environment has 6GB RAM and up to 10GB swap. It has no problem in working when up to five containers with separate FreeIPA deployments are running at the same time in parallel (this is what our CI does). Yet it fails in a simple `ipa-server-upgrade` run which literally stops all services, starts 389-ds instance and runs bunch of `ldapmodify` commands.

**Comment by abbra** [comment]
This particular container had the following memory statistics:
```
1_master_1: /sys/fs/cgroup/memory/memory.memsw.failcnt=0
1_master_1: /sys/fs/cgroup/memory/memory.memsw.limit_in_bytes=3460300800
1_master_1: /sys/fs/cgroup/memory/memory.memsw.max_usage_in_bytes=2309222400
1_master_1: /sys/fs/cgroup/memory/memory.failcnt=1298
1_master_1: /sys/fs/cgroup/memory/memory.max_usage_in_bytes=2306867200
1_master_1: /sys/fs/cgroup/memory/memory.limit_in_bytes=2306867200
1_master_1: /proc/sys/vm/swappiness=60
```

E.g. it had 3GB RAM allocated and never exceeded 2.3GB usage.

I'm attaching `grep bdb errors` output from 389-ds errors log for the lifetime of this test. It starts from an instance setup.
[bdb_errors.txt](https://github.com/389ds/389-ds-base/files/12767675/bdb_errors.txt)

**Comment by Firstyear** [comment]
autotune uses the *free* memory for it's tunings because we don't want to stomp on anything that already exists. So if you look you can see that it detects only 346920k (355246080 bytes, 338mb) as available. This might not be 100% accurate, but aligns with the 2.3/3GB numbers. That's likely because everything else freeipa is running already is there in memory and that constrains what DS sees as available. 

So then when the "minimums" kick in at 65536k and you sum them up, you exceed the free memory that we detected which is why it complains. 

Saying this *none* of these are errors - they are only warnings, and the process still proceeds here. Also remember that even though we might detect free memory, we don't account for swap either, and docker swap limits are separately accounted for. 

So simply put, I suspect you need more ram here.

**Comment by abbra** [comment]
Thanks. I tried to increase allocation of RAM for that test and it passed in a first go. I'll run it several times to see what is better...

**Comment by droideck** [comment]
Closing. Please reopen if the issue still persists.
